js/ui/appIconBar.js

```diff
@@ -676,6 +690,24 @@ const ScrolledIconList = new Lang.Class({
             }));
     },
 
+    _getAppButtons: function() {
+        return this._taskbarApps.values().filter(function (value) {
+            // Excluded apps have a null button
+            return value != null;
+        });
+    },
+
+    getNumAppButtons: function() {
+        return this._getAppButtons().length;
+    },
+
+    activateNthApp: function(index) {
+        let appButton = this._getAppButtons()[index];
```

Comment:
If `getNumAppButtons()` goes away, you can probably change the check below to be `if (!appButton)` and remove the `_getAppButtons()` function too.